src/main/java/com/linkedin/kmf/services/ConsumeService.java

```diff
@@ -131,7 +131,7 @@ public void run() {
   }
 
   private void consume() throws Exception {
-    Map<Integer, Long> nextIndexes = new HashMap<>();
+    Map<String, Long> nextIndexes = new HashMap<>();
```

Comment:
Why do we change the key from partition number to record's key? It seems to me that the original code would work without change, no?